packages/form/src/component/number/context/NumberContext.ts

```diff
@@ -0,0 +1,12 @@
+import { createContext } from "react";
+
+export interface NumberContextProps {
+  locale: string;
+  currency?: string;
+}
+
+export const NumberContext = createContext<NumberContextProps>({
+  locale: "en-IN",
+});
+
+export const NumberLocaleProvider = NumberContext.Provider;
```

Comment:
Instead of using `NumberContext.Provider` as public api to update the context I would prefer a react component that implements the interface `NumberContextProps`. See `FormAdapter.tsx` for an example.  `NumberContext` should be used only internally.

Reply:
Refactored as suggested + better naming: NumberLocaleProvider => NumberI18nProvider